scenarioo-docu-generation-example/src/test/java/org/scenarioo/uitest/example/infrastructure/ScenarioDocuWritingRule.java

```diff
@@ -91,7 +89,7 @@ protected void starting(final Description testMethodDescription) {
 	private Scenario createScenario(final Description testMethodDescription) {
 		Scenario scenario = new Scenario();
 		String description = "";
-		String name = createScenarioName(testMethodDescription);
+		String name = ScenarioConfiguration.getScenarioName(testMethodDescription, MultipleBuildsRule.getCurrentBuildRun());
```

Comment:
This class `ScenarioDocuWritingRule` was once meant as a real example, how it could look in a real project for generating scenario data for tests by intercepting them. With the change to now modify the test names in some cases depending on the build run, is a special case just for dummy demo data generation. That is why I think we should make this more explicit, by following changes: 1. Rename class "ScenarioConfiguration" to "MultipleBuildsDummyTestNameGenerator" 2. document the class that this is only needed for dummy data generation and would never be part of a real `ScenarioDocuWritingRule`. 3. Let the generator class only implement that part of the name genertion that is not meant to be done in a real scenario rule inside a method called something like `changeTestNameForSomeBuildsForDummyDemoData` 4. then put the old logic for name generation back into the Rule, and call the dummy name generator to change the generated usual name only for some builds. 
This would make it better understandable here, what is going on and waht is usual code and what code is only there for special demo data generation.